### ISSUE TITLE
Unit tests are not working for .NET Core

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonSDK#29]
Visual Studio is not able to run any .NET Core targeted unit tests for some reason.  .NET Framework tests do work though.

I'm seeing this in the test logs:
```
========== Starting test discovery ==========
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.5+1caef2f33e (64-bit Desktop .NET 4.0.30319.42000)
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.5+1caef2f33e (64-bit Desktop .NET 4.0.30319.42000)
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.5+1caef2f33e (64-bit Desktop .NET 4.0.30319.42000)
[xUnit.net 00:00:01.01]   Discovering: Test.Neon.Common
[xUnit.net 00:00:01.21] Skipping: Test.Neon.Common (Could not find/load any of the following assemblies: xunit.execution.desktop.dll)
[xUnit.net 00:00:01.20] Skipping: Test.Neon.Common (Could not find/load any of the following assemblies: xunit.execution.desktop.dll)
No test is available in C:\src\neonSDK\Test\Test.Neon.Common\bin\Debug\netcoreapp31\Test.Neon.Common.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
No test is available in C:\src\neonSDK\Test\Test.Neon.Common\bin\Debug\net5.0\Test.Neon.Common.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
[xUnit.net 00:00:01.46]   Discovered:  Test.Neon.Common
========== Test discovery finished: 692 Tests found in 2.2 sec ==========
Building Test Projects
Starting test discovery for requested test run
========== Starting test discovery ==========
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.5+1caef2f33e (64-bit Desktop .NET 4.0.30319.42000)
[xUnit.net 00:00:00.14] Skipping: Test.Neon.Cryptography (Could not find/load any of the following assemblies: xunit.execution.desktop.dll)
No test is available in C:\src\neonSDK\Test\Test.Neon.Cryptography\bin\Debug\net6.0\Test.Neon.Cryptography.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
========== Test discovery finished: 0 Tests found in 905.7 ms ==========
========== Starting test run ==========
========== Test run finished: 0 Tests (0 Passed, 0 Failed, 0 Skipped) run in < 1 ms ==========
```

I poked around to xunit code on GitHub a bit, and it looks like **xunit.core** believes that a .NET Framework targeted assembly is being tested, so it's looking for the **xunit.execution.desktop.dll** in the **bin** folder instead of **xunit.execution.dotnet.dll** which is the .NET Core runner.  It's not clear why this is happening.

**Comment by jefflill** [comment]
I'm setting this aside for a few hours and will have another look later.

**Comment by jefflill** [comment]
Perhaps this is related: https://developercommunity.visualstudio.com/t/Visual-Studio-does-not-discover-or-run-x/10154061?space=8&q=xunit&sort=newest

** posted a week or so ago: 9/21/2022

**Comment by jefflill** [comment]
Fixed this a while ago.  The problem was that we disabled `TargetFrameworkAttribute` generation to avoid duplicate symbol conflicts and those are required for unit test discovery and debugging so that the type of the assembly can be determined. by those tools.

The final fix was to add the `BuildInfo` projects in the different repos to handle `TargetFrameworkAttribute` via shared file links and also to wrap `GitInfo` to avoid duplicate symbols from that too.
